src/ReasonReact.rei

```diff
@@ -254,6 +254,8 @@ module Router: {
       for an example.
       */
   let dangerouslyGetInitialUrl: unit => url;
+  /** hook for watching url changes */
+  let useUrl: unit => url;
```

Comment:
Offline you were mentioning that you think we should move this into a separate module to get the dependencies correct. How exactly were you thinking about doing that? Want to make sure we have a plan for this before we add more stuff here.

Reply:
IMO the `ReasonReact.re` as it is now on the hooks branch should be split into three parts:

1. "Classic" ReasonReact (record-based) components
2. The router
3. The `useRecordApi` compatibility hook

If you don't want a breaking change, we could move the contents of the `Router` submodule into `ReasonReactRouter.re` and have `module Router = ReasonReactRouter;` in `ReasonReact.re`.

`useRecordApi` could be moved into `ReasonReactRecordApi.re` or `ReasonReactCompat.re` or something like that.

This way new projects using the hooks API only can just use `React.re` and, if they need it, `ReasonReactRouter.re` without pulling in "legacy" stuff.

But this can be done in a separate PR after `useUrl` is added.

Reply:
Got it. Yeah I like `ReasonReactCompat.re` and `ReasonReactRouter.re` with a module alias for backwards compat. Kind of a shame about the collision with ReactRouter. Let’s go with this and talk to Cheng in the next PR about where he wants that one.